### ISSUE TITLE
Support Mocha 8

### ISSUE THREAD

**Issue by Prophet32j** [stryker-mutator/stryker-js#2274]
**Question**

Are there plans to support Mocha 8?  
I think I have issues with Mocha 8/Stryker 3.2.4 together. My code is well covered with boundaries and conditionals. All mutants survive. Even on module imports being String-literaled. I'm guessing this is a Mocha 8 issue since my other project with Mocha 7 and Stryker 2.5 working.

**Stryker environment**

```
+-- @stryker-mutator/core@3.2.4
+-- @stryker-mutator/javascript-mutator@3.2.4
```

```
+-- mocha@8.0.1
```

**Additional context**

**Comment by nicojs** [comment]
Hi 👋  @Prophet32j  

Could you try v3.3 of Stryker? We officially support mocha 8 since 3.3:

https://github.com/stryker-mutator/stryker/blob/epic/mutation-switching/CHANGELOG.md#330-2020-06-16

I'm missing `@stryker-mutator/mocha-runner` in your list of dependencies. Is this deliberate? If so, you're probably using the command test runner. That one should have worked with mocha 8 all along 😲

**Comment by maenu** [comment]
I just updated to stryker v3.3.1 and it gives me:

```
npm WARN @stryker-mutator/mocha-framework@3.3.1 requires a peer of mocha@>= 2.3.3 < 8 but none is installed. You must install peer dependencies yourself.
```

The peer-dependency of mocha-framework indeed still refers to mocha < 8:

```
"peerDependencies": {
  "@stryker-mutator/core": "^3.0.0",
  "mocha": ">= 2.3.3 < 8"
}
```

While mocha-runner allows mocha < 9:

```
"peerDependencies": {
  "@stryker-mutator/core": "^3.0.0",
  "mocha": ">= 2.3.3 < 9"
}
```

**Comment by Prophet32j** [comment]
This is what I had, and my tests would not pass any mutations, even though
I know my tests account for them.

**Comment by Prophet32j** [comment]
Ok. I have figured out why my tests would not run. Stryker doesn't like `module-alias` when importing modules.

```javascript
const matchRelationships = require('../../src/er-studio/match-relationships')
```
vs
```javascript
const matchRelationships = require('@src/er-studio/match-relationships')
```

For any tests that would use the aliased module reference, those tests would either time out or all mutants would live.

If there's no ability to load aliased modules, there should be some sort of NOTICE on the README to inform developers of that limitation.

Either that or I didn't correctly inform Stryker of aliased modules, but I fail to find any documentation on that.

**Comment by Prophet32j** [comment]
@nicojs

**Comment by nicojs** [comment]
Yeah, module aliassing does not play nice with Stryker. Allowing Stryker to mutate your code in-place will solve this issue, see #2163 .

> Stryker doesn't like `module-alias` when importing modules

I'm not an expert on this subject. What is `module-alias` exactly? How does it work? If I knew what it was and how it works, I might be able to help you better here.

> Either that or I didn't correctly inform Stryker of aliased modules, but I fail to find any documentation on that.

True, we're missing more extensive documentation, or FAQ for this. Please feel free to add it! I think the stryker-handbook might be the place for this documentation https://github.com/stryker-mutator/stryker-handbook/tree/master/stryker

**Comment by profplum700** [comment]
I'm trying to get Stryker 4.1.2 running on a limited set of our unit tests. The Terminal output instruments the tests, starts the test run then produces an error that `No tests were executed`. Is this because our version of Mocha isn't supported, similar to [this](https://github.com/stryker-mutator/stryker/issues/1937)?

Here is a screenshot from stryker.log:
![image](https://user-images.githubusercontent.com/13546520/99974365-b8954000-2d98-11eb-9891-190a24545da9.png)

This is my stryker.conf.json:
`{
  "$schema": "./node_modules/@stryker-mutator/core/schema/stryker-schema.json",
  "packageManager": "npm",
  "reporters": [
    "html",
    "clear-text",
    "progress"
  ],
  "testRunner": "mocha",
  "coverageAnalysis": "perTest",
  "mutate": [
    "v3/SquaredUp.App/PristinePackages/SCOM/Pages/**/*.js"
  ],
  "fileLogLevel": "trace"
}`

I am able to kick off all our unit tests using `npm test`. This has got me stumped.

I'm pretty new to this kind of thing so please let me know what other info you need and I'll get it!

**Comment by nicojs** [comment]
@profplum700 could you please try again with the new `--inPlace` mode: https://stryker-mutator.io/docs/stryker/configuration/#inplace-boolean (just released in Stryker 4.4). This should solve your issues.

**Comment by profplum700** [comment]
> @profplum700 could you please try again with the new `--inPlace` mode: https://stryker-mutator.io/docs/stryker/configuration/#inplace-boolean (just released in Stryker 4.4). This should solve your issues.

Thanks for the heads up. I won't be able to test it out as I've moved onto other work but I've made a note of it for the future.

**Comment by nicojs** [comment]
Mocha 8 is since supported.
